### PR TITLE
Support INSERT statement over SQL adapter

### PR DESCRIPTION
**Pull request by aljazerzen**
Todo and general implementation strategy tracked in #7294

### CODE REVIEW

File: edb/pgsql/resolver/command.py [hunk 1]
```diff
@@ -90,3 +105,712 @@ def resolve_CopyStmt(stmt: pgast.CopyStmt, *, ctx: Context) -> pgast.CopyStmt:
...
+        val_col_pg = pg_res_expr.resolve_column_kind(
+            val_table, val_col.kind, ctx=ctx
+        )
+        if is_link:
+            val_col_pg = pgast.TypeCast(
+                arg=val_col_pg, type_name=pgast.TypeName(name=('uuid',))
+            )
+        value_target_list.append(pgast.ResTarget(name=ptr_name, val=val_col_pg))
+
+    # source needs an identity column, so we need to invent one
```
Comment (msullivan) [inline edb/pgsql/resolver/command.py:643]: iterator column, not identity, right?

**Review by msullivan — APPROVED** [review]
Great work!

Could you make a list of all the TODOs and put it in the issue?

File: edb/pgsql/resolver/command.py [hunk 2]
```diff
@@ -90,3 +105,712 @@ def resolve_CopyStmt(stmt: pgast.CopyStmt, *, ctx: Context) -> pgast.CopyStmt:
...
+        ptr_name = ptr.get_shortname(ctx.schema).name
+
+        # prepare pre-projection of this pointer value
+        val_col_pg = pg_res_expr.resolve_column_kind(
+            val_table, val_col.kind, ctx=ctx
+        )
+        if is_link:
+            val_col_pg = pgast.TypeCast(
+                arg=val_col_pg, type_name=pgast.TypeName(name=('uuid',))
+            )
```
[the remark this answers is not in the document]
Comment (aljazerzen) [reply edb/pgsql/resolver/command.py:640]: To produce a better error when users are trying to insert non-uuids into link columns.

For example:

```
edgedb=# INSERT INTO "Document" (owner_id) VALUES (4);
ERROR:  cannot cast type integer to uuid
LINE 1: INSERT INTO "Document" (owner_id) VALUES (4);
        ^
```

Not great, but better than talking about "operator `=` not existing for uuid and integer".

File: edb/pgsql/resolver/command.py [hunk 3]
```diff
@@ -90,3 +105,712 @@ def resolve_CopyStmt(stmt: pgast.CopyStmt, *, ctx: Context) -> pgast.CopyStmt:
...
+            stmt.relation.alias.aliasname,
+            ctx,
+        )
+    else:
+        res_query = pgast.SelectStmt()
+        res_table = context.Table()
+
+    if not res_query.ctes:
+        res_query.ctes = []
+    res_query.ctes.extend(pg_res_rel.extract_ctes_from_ctx(ctx))
```
[the remark this answers is not in the document]
Comment (aljazerzen) [reply edb/pgsql/resolver/command.py:682]: No. `extract_ctes_from_ctx` returns a list of ctes, but only when ctx is top-level.

This is here to achieve exactly the opposite: have all generated CTEs at the top-level.

**Pull request merged**